src/Cooked/MockChain/MinAda.hs

```diff
@@ -0,0 +1,59 @@
+-- | This module provides functions to ensure outputs are furnished with enough
+-- ada to satisfy the minimum ada constraint, by themselves or within skeletons.
+module Cooked.MockChain.MinAda
+  ( toTxSkelOutWithMinAda,
+    toTxSkelWithMinAda,
+    getTxSkelOutMinAda,
+  )
+where
+
+import Cardano.Api qualified as Cardano
+import Cardano.Api.Ledger qualified as Cardano
+import Cardano.Api.Shelley qualified as Cardano
+import Cardano.Ledger.Shelley.Core qualified as Shelley
+import Cardano.Node.Emulator qualified as Emulator
+import Cardano.Node.Emulator.Internal.Node.Params qualified as Emulator
+import Control.Monad.Except
+import Cooked.MockChain.BlockChain
+import Cooked.MockChain.GenerateTx
+import Cooked.Skeleton
+import Cooked.ValueUtils
+import Optics.Core
+import Plutus.Script.Utils.Ada qualified as Script
+
+-- | This provides the minimum amount of ada required in a given `TxSkelOut`. As
+-- we need to transform our output into a Cardano output to compute this value,
+-- this function can fail.
+getTxSkelOutMinAda :: Emulator.Params -> TxSkelOut -> Either GenerateTxError Integer
+getTxSkelOutMinAda Emulator.Params {..} txSkelOut =
+  Cardano.unCoin
+    . Shelley.getMinCoinTxOut pEmulatorPParams
+    . Cardano.toShelleyTxOut Cardano.ShelleyBasedEraConway
+    . Cardano.toCtxUTxOTxOut
+    <$> generateTxOut pNetworkId txSkelOut
+
+-- | This transforms an output into another output which necessarily contains at
+-- least the minimal required ada. If the previous quantity of ada was
+-- sufficient, it remains unchanged. This requires an iterative process, as
+-- adding ada into an output can potentially increase its size and thus make it
+-- require more minimal ada (although this remains to be witnessed in practice).
+-- This approach was inspired by
```

Comment:
This is a good question. From my experience (I tried with dummy outputs in a repl with various configurations), changing only the ada in an output does not change it required min ada so this function will always succeed in the first go and not even go into second pass. But this is empirical and I cannot be 100% sure. Since min ada depend on the size of the output, I assume that the number of ada it contains only always takes a single byte of memory and thus increasing the ada amount will never increase the required min ada.